packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Parses a number and unit from a value.
+ *
+ * @param  toParse Value to parse
+ *
+ * @return  The extracted number and unit.
+ */
+export function parseUnitValue(
+	toParse: string | number
+): [ number | string | undefined, string | undefined ] {
+	// eslint-disable-next-line eqeqeq
+	if ( toParse == null ) {
+		return [ undefined, undefined ];
+	}
+
+	const value = String( toParse ).trim();
```

Comment:
Should we replace all instances of white space instead of just trimming the start and the end?

Reply:
Actually, probably not necessary if considering that currently the RegExp used below accounts for `\s*`